src/sidebar/util/test/fake-window.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+class FakeWindow {
```

Comment:
This fake implementation of `Window` was moved from oauth-auth-test.js mostly unchanged. It is mostly used by the `OAuthClient` tests but is still used by the authorization service tests that check for handling of "storage" events that occur when the user has the client open in multiple tabs and changes their login in one of them.